### ISSUE TITLE
THROWS_METHOD_THROWS_RUNTIMEEXCEPTION from return value

### ISSUE THREAD

**Issue by nmatt** [spotbugs/spotbugs#3919]
Consider:
```java
public void method() throws RuntimeExceptionTypeA, RuntimeExceptionTypeB
{
    try
    {
        // something
    }
    catch (SpecificCheckedException ex)
    {
        throw mapException(ex);  // should not cause a warning
    }
}

private RuntimeException mapException(Exception ex)
{
    if (someSpecificState)
    {
        return new RuntimeExceptionTypeA(ex);
    }
    else
    {
        return new RuntimeExceptionTypeB(ex);
    }
}
```
This code throws different runtime exception types depending on some condition. The code that decides on which exception to throw is factored out into a separate method (for example because it is used in multiple places). This should not cause a warning, the same as the following functionally equivalent code doesn't cause a warning:
```java
public void method() throws RuntimeExceptionTypeA, RuntimeExceptionTypeB
{
    try
    {
        // something
    }
    catch (SpecificCheckedException ex)
    {
        if (someSpecificState)
        {
            throw new RuntimeExceptionTypeA(ex);
        }
        else
        {
            throw new RuntimeExceptionTypeB(ex);
        }
    }
}
```
I would argue that a warning is only warranted when the exception that is thrown can be shown to be the result of a constructor invocation of the class `RuntimeException`, but not of any runtime exception subtype.
